File: models/movie.go
package models

import "time"

type Movie struct {
	ID          int        `json:"id"`
	Title       string     `json:"title"`
	Genre       string     `json:"genre"`
	ReleaseDate time.Time  `json:"release_date"`
	EndDate     time.Time  `json:"end_date"`
	IsShowing   bool       `json:"is_showing"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	DeletedAt   *time.Time `json:"deleted_at,omitempty"`
}

File: src/main/kotlin/com/example/moviereviewapp/domain/Movie.kt
package com.example.moviereviewapp.domain

import com.example.moviereviewapp.dto.MovieDTO
import jakarta.persistence.*
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
data class Movie(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    var title: String,
    @Enumerated(EnumType.STRING)
    var genre: Genre,
    var releaseDate: LocalDate,
    var endDate: LocalDate,
    var isShowing: Boolean,
    val registrationDate: LocalDate = LocalDate.now(),
    var isDeleted: Boolean = false,

    @Column(name = "updated_date")
    var updatedDate: LocalDateTime? = null
) {
    fun toDTO(): MovieDTO {
        return MovieDTO(
            id = id,
            title = title,
            genre = genre,
            releaseDate = releaseDate,
            endDate = endDate,
            isShowing = isShowing,
            registrationDate = registrationDate
        )
    }
}